page-manager.js

```diff
@@ -299,157 +310,162 @@ define([
 
             return (controllerIndex);
         },
+
         /**
          * This function will handle all the routing request and will process accordingly
          * and process the page transition
+         * @todo refactor this method, is too big
          */
         _handleRoute : function (controllerInfo, urlParams, options) {
             var self = this;
-            log('Handling route...', arguments);
-            // Check if the specified controller exists
-            // Check if the specified controller has been instanciated
-            // Check if the specified controller was loaded
-            // We try to resolve the specified controller path to an object
-            var controllerPath = controllerInfo.controller,
-                routeId = controllerInfo.routeId,
+            /** runs all the defined middlewares for the route:before phase */
+            self.middleware.run(Middleware.PHASES.ROUTE, Middleware.SUBPHASES.BEFORE, controllerInfo, function(result) {
```

Comment:
I Added a default middleware instance to the PageManager in case the app didn't define one.